### DISCUSSION TITLE
Cannot invite more than 10 users to organisation

### DISCUSSION THREAD

**Discussion by the-marat** [dani-garcia/vaultwarden#4707] [Q&A]
Is there any limits for user number in vaultwarden self hosted???

Because cannot add more than 10 users to oraganisation - when press INVITE MEMBER for 11th user it shows warning:

> Upgrade organization
> Free organizations may have up to 10 members. Upgrade to a paid plan to invite more members.

after pressing UPGRADE button it redirects to #/vault?upgrade=true where I can see a main page of vault...

Versions

Server Installed Update     1.30.5-a4c7fadb 
Server Latest    1.30.5-fda77afc 
Web Installed Ok     2024.5.1 
Web Latest     2024.5.1 
Database    SQLite: 3.45.0

**Comment by stefan0xC** [comment]
Thanks for the report. I'll take a look if we can fix that.

**Reply by stefan0xC** [reply]
should be fixed by https://github.com/dani-garcia/bw_web_builds/pull/171

**Reply by Nerevar-de** [reply]
> should be fixed by [dani-garcia/bw_web_builds#171](https://github.com/dani-garcia/bw_web_builds/pull/171)

I pulled fresh on July 11th and have the same message upon inviting a user.

**Reply by BlackDex** [reply]
It is fixed in the `testing` tagged version. We have not yet released a patch version.

**Reply by Nerevar-de** [reply]
> We have not yet released a patch version.

Any ETA for that?

**Reply by BlackDex** [reply]
Nope. We do not have those.

**Reply by scolastico** [reply]
Stuff like that should never go to production. Its a shame that this has gotten through, and not even a hotfix release is planned.

**Reply by BlackDex** [reply]
It is planned in that sense that we want to fix other stuff too.
If you want to prevent these items, then use testing and report issues to us so we can fix them. This still is a open source project done in the spare time of the maintainers.

If you have issues with that, use Bitwarden.

**Reply by Nerevar-de** [reply]
> Stuff like that should never go to production. Its a shame that this has gotten through, and not even a hotfix release is planned.

As @ROXASBrandon mentioned in https://github.com/dani-garcia/vaultwarden/discussions/4707#discussioncomment-10014237 , that check only applies while having exactly 10 members in your organization. Just remove a member and add two at the same time. After reaching the mark of 11 you can invite infinite members 1 by 1 again.

**Comment by the-marat** [comment]
Maybe it can help. I use this manual https://community.hetzner.com/tutorials/how-to-install-vaultwarden at debian 12

Not DOCKER

**Reply by the-marat** [reply]
Made update to 

> Versions
> 
> Server Installed Ok
>     1.31.0 
> Server Latest
>     1.31.0 
> Web Installed Ok
>     2024.5.1b 
> Web Latest
>     2024.5.1b 
> Database
>     SQLite: 3.45.0 


And the same error (((


>  Upgrade organization
> Free organizations may have up to 10 members. Upgrade to a paid plan to invite more members.

**Comment by BlackDex** [comment]
fyi, this should be fixed in the latest Vaultwarden v1.31.0 release, which includes the web-vault v2024.5.1b which has the needed patches. Since this project does not control all other deployment methods besides the containers we build, we can not ensure other delivery methods are up-to-date with there packages.

**Reply by the-marat** [reply]
i've updated all that manually and system admin/diagnostics shows me lates versions  2024.5.1b and  1.31.0

**Reply by the-marat** [reply]
I've just created new organisation and invited 9 new users - and after 10th users it shows warning again. So it's not problem of old organisation .. for new also

**Reply by stefan0xC** [reply]
Inviting 15 users at once works for me with the `latest` image (both on existing and new organizations). Maybe you need to refresh your browser cache?

**Reply by the-marat** [reply]
no. browser cache is not a problem - tried 3 differend browser and incognito mode.

I've just created new VPS at Debian 12. Install from scratch all - the problem not resolved.. I can give you full access to that VPS in private message.

Here is my installation path (not docker)

```
apt update
apt upgrade
apt install curl
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
source $HOME/.cargo/env
apt update && apt install -y build-essential git libssl-dev pkg-config libsqlite3-dev nginx certbot
mkdir ~/source && cd ~/source
git clone https://github.com/dani-garcia/vaultwarden.git # clones the repo
cd vaultwarden                                           # changes directory to vaultwarden
cargo build --features sqlite --release                  # builds vaultwarden with SQLite backend
mkdir -p /var/lib/vaultwarden && cd /var/lib/vaultwarden
mkdir -p data
wget https://raw.githubusercontent.com/dani-garcia/vaultwarden/main/.env.template
mv .env.template .env
cd /var/lib/vaultwarden
wget https://github.com/dani-garcia/bw_web_builds/releases/download/v2024.5.1b/bw_web_v2024.5.1b.tar.gz
tar -xvf bw_web_v2024.5.1b.tar.gz 
rm bw_web_v2024.5.1b.tar.gz 
nano /var/lib/vaultwarden/.env
cp ~/source/vaultwarden/target/release/vaultwarden /usr/local/bin/vaultwarden
chmod +x /usr/local/bin/vaultwarden
useradd -m -d /var/lib/vaultwarden vaultwarden
chown -R vaultwarden:vaultwarden /var/lib/vaultwarden
nano /etc/systemd/system/vaultwarden.service # use the configuration below
systemctl daemon-reload
systemctl enable vaultwarden
systemctl start vaultwarden
cd /etc/nginx/sites-available
nano vaultwarden.conf          # use the configuration below
ln -s /etc/nginx/sites-available/vaultwarden.conf /etc/nginx/sites-enabled/vaultwarden.conf
nginx -t                       # check if the configuration is valid
apt install python3-certbot-nginx
certbot --nginx                # make sure to choose redirect to https option
```

**Reply by BlackDex** [reply]
I just invited 20 users, `user1@example.tld,user2@example.tld,user3@example.tld,user4@example.tld,user5@example.tld,user6@example.tld,user7@example.tld,user8@example.tld,user9@example.tld,user10@example.tld,user11@example.tld,user12@example.tld,user13@example.tld,user14@example.tld,user15@example.tld,user16@example.tld,user17@example.tld,user18@example.tld,user19@example.tld,user20@example.tld` and that worked just fine without any issues.

**Reply by Masgalor** [reply]
I can confirm something is wrong with the webvault from the [release-page](https://github.com/dani-garcia/bw_web_builds/releases/download/v2024.5.1b). It is definitely missing the fix for the org limit.

**Reply by BlackDex** [reply]
@Masgalor Then i would suggest to check again as it really is there:
 - https://github.com/dani-garcia/bw_web_builds/compare/v2024.5.1...v2024.5.1b#diff-7fed509d522fad2ee9d51a97d7513edea67f9691808a87b3f9f4058b3a9aa97cR34
 - https://github.com/dani-garcia/bw_web_builds/blob/26c122f20aeb921cb22eef7dfc3add6d20ed3582/patches/v2024.5.0.patch#L34
 - https://github.com/dani-garcia/bw_web_builds/actions/runs/9846631651/job/27184801493#step:8:285

**Reply by Masgalor** [reply]
Im sorry, you are right, the problem does also exist if I extract the files from the docker image.

However, I can reproduce the problem @the-marat is talking about.
I can add (via bulk) as many members as I want, no problem.
But if there are exact 10 members in my org, I can not add any more.
I have to remove one (to get to 9) and invite two others at once (so I get to 11) now I can invite more members as long as I stay above 10.

**Reply by BlackDex** [reply]
Ok, i just did the following.

```bash
docker run --rm -it -e DISABLE_ADMIN_TOKEN=true -e I_REALLY_WANT_VOLATILE_STORAGE=true -e LOG_LEVEL=debug -e ROCKET_PORT=8080 --network=host vaultwarden/server:latest-alpine
```

Logged-in via `http://127.0.0.1:8080`, created an account, created an org, invited users.

And i can invite 20 users in one go, and still add a 21th user after that (22nd actually).
![image](https://github.com/dani-garcia/vaultwarden/assets/610450/fc69570c-9bcd-4bec-b998-e307873244a3)

There is nothing wrong with the latest release of the web-vault and the container images in regards to adding more then 10 users to an org.

You either are using a wrong web-vault version, or something in the browser cache, or reverse proxy cache is serving old data.
Else, we need more details, but a clean environment works as it should.

**Reply by BlackDex** [reply]
Please provide a step-by-step item, what you do, as it seems to be working fine for me.

**Reply by BlackDex** [reply]
Oh, wait, if you have exactly 10 users it triggers this. It doesn't if you go beyond 10. Weird behavior.
That is strange, didn't noticed this specific before.

**Reply by Masgalor** [reply]
You dont seem to understand that this problem is with the opening dialog of the `invite member button`.
This is due to my poor language skills, I will make a video when Im at home.

1. Create a new org
Your org has now one member (yourself)

2. Add 9 other members, not 5, not 20, exact 9 others so you have overall 10 members.
3. Now hit the `invite member button`, you will **not** be allowed to add more members.
4. Remove one of your members, so you will have overall 9 members.
5. Now hit the `invite member button`, you will be allowed to add more members.
6. Add 2 more members, so you have overall 11 members.
7. Now hit the `invite member button`, you will still be allowed to add more members.

**Reply by BlackDex** [reply]
I understand already, as you can see me previous comment :).

**Reply by stefan0xC** [reply]
Ah, thank for the writeup. I've found the specific check and added a patch for it: https://github.com/dani-garcia/bw_web_builds/pull/172

**Reply by the-marat** [reply]
I wrote above that for experiment I created clean Debian 12 VPS and installed all from latest repos.. version in ADMIN panel showed as latest - and the problem is still exist.. I created new organisation and manually tried to add users - and can't add more than 10 if I add 1 user per invite. 

BUT I can add several (bulk) users in 1 invitation and it really works.. after users more than 11 - I can manually add by 1 user again.. 

If I have 20 users and delete 10 users to leave 10 active users - I again has a problem when adding users by 1 per invite..

So problem with 10 users only.. you just need to jump over with bulk invivtation to pass this glitch.

P.S. didn't see all comments - so can repeat previos replies))

**Reply by stefan0xC** [reply]
After looking into it a bit and talking with @BlackDex about the issue we should probably set either these to higher numbers: https://github.com/dani-garcia/vaultwarden/blob/d04b94b77d95a9968f61f71969f65aaaa8a496d6/src/db/models/organization.rs#L372-L373 and/or change the returned `productTierType` of the organization to `3` (i.e. `Enterprise` instead of `Free`)so the checks are skipped instead of me patching them out in the web-vault.

**Reply by dtkbrbq** [reply]
Same message while trying to create more than 10 collections in organization.

**Reply by ROXASBrandon** [reply]
@dtkbrbq the workaround for now is to have 9 users, and then add two at the same time to get around that check for 10 users.
![image](https://github.com/dani-garcia/vaultwarden/assets/1660734/659a9b69-ecdb-4aef-9e5c-c234b5e9541a)

**Reply by BlackDex** [reply]
It is fixed already in the `testing` tagged images btw.

**Reply by thepromisedev** [reply]
This limitation can be bypassed as the colleague mentioned by sending everything in one invitation separated by ","; 
however, the issue persists since no one else can be added afterward.

**Reply by Nerevar-de** [reply]
> [...] however, the issue persists since no one else can be added afterward.

That is not correct. See my comment https://github.com/dani-garcia/vaultwarden/discussions/4707#discussioncomment-10112443 above and also at least 2 others here wrote the same. https://github.com/dani-garcia/vaultwarden/discussions/4707#discussioncomment-9999609 and https://github.com/dani-garcia/vaultwarden/discussions/4707#discussioncomment-9999488 for example.

The (fault) check seems to only fail at exactly 10 org members.

**Reply by thepromisedev** [reply]
is correct only at 10,  sorry
